cmd/sensors/src/lib.rs

```diff
@@ -302,6 +325,179 @@ fn print(
             errs = vec![None; rval.len()];
         }
 
+        Ok(SensorData { values: rval, errors: errs })
+    }
+}
+
+/// Reader for sensor data that pulls it out of RAM directly
+struct RamSensorReader {
+    /// Position of `LAST_READING` in global RAM
+    last_reading: HubrisVariable,
+    last_reading_buf: Vec<u8>,
+
+    /// Position of `DATA_VALUE` in global RAM
+    data_value: HubrisVariable,
+    data_value_buf: Vec<u8>,
+
+    /// Position of `NERRORS` in global RAM
+    nerrors: HubrisVariable,
+    nerrors_buf: Vec<u8>,
+
+    /// Indexes of sensors that we care about
+    ///
+    /// We'll read the entire array (for simplicity) then filter just the ones
+    /// that matter
+    sensors: Vec<usize>,
+}
+
+impl RamSensorReader {
+    fn new(
+        hubris: &HubrisArchive,
+        sensors: &[(usize, HubrisSensor)],
+    ) -> Result<Self> {
+        let find_var = |v| {
+            hubris
+                .qualified_variables()
+                .find(|&(n, _v)| n == format!("task_sensor::main::{v}"))
+                .ok_or_else(|| anyhow!("could not find DATA_VALUE"))
```

Comment:
this should probably be 
```suggestion
                .ok_or_else(|| anyhow!("could not find {v}"))
```
right? presently, it will _always_ say that it could not find `DATA_VALUE` even if it was actually `LAST_READING` or `NERRORS` that couldn't be found...

Reply:
Good catch, fixed in 019575ef3dee0c0b9787019f0ce18bf5beb8648d